tests/scripts/check_test_cases.py

```diff
@@ -111,6 +112,23 @@ def walk_ssl_opt_sh(self, file_name):
                 self.process_test_case(descriptions,
                                        file_name, line_number, description)
 
+    def walk_compat_sh(self, file_name):
+        """Iterate over the test cases compat.sh with a similar format."""
+        descriptions = self.new_per_file_state() # pylint: disable=assignment-from-none
+        compat_cmd = ['sh', file_name, '--list-test-case']
+        result = subprocess.run(compat_cmd,
+                                stdout=subprocess.PIPE,
+                                check=False)
+        if result.returncode != 0:
+            print(*compat_cmd, 'returned', str(result.returncode))
+            return
```

Comment:
I think it would be better to print the warnings/errors to `STDERR`.

I'm also considering not to ignore the errors here. Because, if here the condition hit, it would mean that the test script exsits but it has problem listing the cases. I know it would be at an extremely low chance, so I'm not very confident.

Reply:
I agree, we should not ignore the error. (Also agree on printing any warning/error messages to stderr.) Don't use `subprocess.run(check=False)` unless you have a very good reason, and I don't see a good reason here. Unless you're doing something unusual, use `check_output` or `check_call`, and catch the error if you need to.

Reply:
make senses. I used `check_output` to catch the error status.
I think there are two main types of errors.
1. command error. Say that running `compat_cmd` fails because of some exceptions, `check_output` would report the errors. 
```
subprocess.CalledProcessError: Command '['sh', 'tests/compat.sh', '--list-test-case']' returned non-zero exit status 1
```
2. errors inside `compat.sh`. 

eg. `compat;tests/compat.sh: 176: tests/compat.sh: list_test_cas: not found`

If `compat_cmd` is executed successfully, but no expected results are returned. `check_output` won't report this type of error. So I add a comment as a reminder to the option `--list-test-case`. Anyway, this error happens in a very low possibility unless someone writes a bug to the option `--list-test-case`. I think this is out scope of this PR. If we really need to handle this error case, I think we can do it in #5390

Reply:
> 2. errors inside `compat.sh`.
eg. `compat;tests/compat.sh: 176: tests/compat.sh: list_test_cas: not found`

The case is too special. It means `compat.sh` couldn't handle the error correctly. I think we don't need to consider this. The "error" in my comment means "non-zero return value of `compat.sh`". Under this criteria, this case is not actually an "error", but should be seen as "warning". So, the incorrect output of this script should be resulted by mixing up the stderr and stdout [here](
https://github.com/yanrayw/mbedtls/blob/cdc07083345eef2b2404d7a4a9c55c2546a7899f/tests/scripts/check_test_cases.py#L119-L120).

Reply:
14e052f is added to handle more possible errors inside `compat.sh`.